### PR TITLE
[source-mysql-v2] Complete datatypes

### PR DESCRIPTION
**Pull request by xiaohansong**
## What
Completed datatypes. Referring from v1 https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlSourceOperations.java#L99-L162

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt [hunk 1]
```diff
@@ -63,81 +61,70 @@ import jakarta.inject.Singleton
...
-        when (val elementType = type.elementType) {
-            is SystemType -> {
-                val leafType: JdbcFieldType<*> =
-                    leafType(elementType.typeName, elementType.scale != 0)
-                if (leafType == OffsetDateTimeFieldType) {
-                    // Mysql's JDBC driver has a bug which prevents object conversions in
-                    // ArrayDataResultSet instances. Fall back to strings instead.
-                    PokemonFieldType
+    private fun leafType(type: SystemType): JdbcFieldType<*> {
+        val typeName = type.typeName
```
Comment (postamar) [inline airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt:69]: Is it possible to call `MysqlType.getByName(type.typeName)` to get the `MysqlType` enum value and dispatch on that instead?
The resulting `when` dispatch block would look a lot nicer that way.

Comment (postamar) [reply airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt:69]: FYI, style nit: Type declarations are not optional when the type isn't obvious.

I'll look into having this as a linter rule, PRs are better off without these kinds of comments.

File: airbyte-integrations/connectors/source-mysql-v2/src/test/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceDatatypeIntegrationTest.kt [hunk 2]
```diff
@@ -168,9 +168,86 @@ class MysqlSourceDatatypeIntegrationTest {
...
+                "TRUE" to "true",
+                "FALSE" to "false",
+            )
+
+        // Encoded into base64
+        val binaryValues =
+            mapOf(
+                "X'89504E470D0A1A0A0000000D49484452'" to
+                    """"iVBORw0KGgoAAAANSUhEUgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=="""",
+            )
```
Comment (postamar) [inline airbyte-integrations/connectors/source-mysql-v2/src/test/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceDatatypeIntegrationTest.kt:217]: I see a lot more test cases in the original source-mysql, why aren't you porting those over?

**Review by postamar — COMMENTED** [review]
This is looking good. I'd feel better if this included all the test cases from source-mysql. Is there any obstacle to doing this?

File: airbyte-integrations/connectors/source-mysql-v2/src/test/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceDatatypeIntegrationTest.kt (hunk 2 above)
Comment (xiaohansong) [reply airbyte-integrations/connectors/source-mysql-v2/src/test/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceDatatypeIntegrationTest.kt:217]: Good call. I'll port them over.

File: airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt [hunk 3]
```diff
@@ -63,81 +61,70 @@ import jakarta.inject.Singleton
 class MysqlSourceOperations : JdbcMetadataQuerier.FieldTypeMapper, SelectQueryGenerator {
     override fun toFieldType(c: JdbcMetadataQuerier.ColumnMetadata): FieldType =
         when (val type = c.type) {
-            is SystemType -> leafType(c.type.typeName, type.scale != 0)
-            is UserDefinedArray -> ArrayFieldType(recursiveArrayType(type))
-            is UserDefinedType -> PokemonFieldType
+            is SystemType -> leafType(type)
+            else -> PokemonFieldType
```
[the remark this answers is not in the document]
Comment (xiaohansong) [reply airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt:65]: from my understanding this field is just to handle unexpected types, and for those we just takes their raw string value. @postamar can educate me about the name and use!

[the remark this answers is not in the document]
Comment (theyueli) [reply airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt:65]: @postamar shall we rename it? This is kind of hard to understand...

File: airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt (hunk 1 above)
Comment (xiaohansong) [reply airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt:69]: nice. thanks!

File: airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (postamar) [reply airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlSourceOperations.kt:66]: Yes. Any ideas for a better name? `CatchAllFieldType`? `FallbackFieldType`? `StringifiedFieldType`?

**Review by postamar — COMMENTED** [review]
@xiaohansong sorry for not giving this another look sooner. I didn't look at the test cases in detail, are they all there? If not, that shouldn't block merging this PR, but it would warrant an issue tracking porting over what's missing.

**Review by postamar — APPROVED** [review]

**Pull request merged**